File: 240128/괄호 쌍 만들어주기 2/pair-parentheses-2.py
a=list(str(input()))

ans=0

n=len(a)

for i in range(n-3):
    for j in range(i+2,n-1):
        if a[i]==a[i+1]=='(' and a[j]==a[j+1]==')':
            ans+=1
        

print(ans)

File: 260106/괄호 쌍 만들어주기 2/pair-parentheses-2.js
const fs = require("fs");
const input = fs.readFileSync(0).toString().trim().split('\n');
const a = input[0];
// Please Write your code here.

const n = a.length;

let ans=0;

for(let i=0; i<n; i++){
    for(let j=i+2; j<n-1; j++){
        if(a[i]==='(' && a[i+1]==='(' && a[j]===')' && a[j+1]===')'){
            ans++
        }
    }
}

console.log(ans)